app/controllers/puzzles_controller.rb

```diff
@@ -10,10 +10,18 @@ def index
     @puzzles = @producer.puzzles.includes(:rates).order(:name) if @producer
     @puzzles = @inventor.puzzles.includes(:rates).order(:name) if @inventor
     @pre_set = @material || @producer || @inventor
+
     if !@pre_set
-      @puzzles = Puzzle.search((params.dig(:puzzle, :q) || '*'),
-        fields: [:name, :inventor, :producer], misspellings: { below: 5 },
-         match: :word_middle, order: :name)
+      @q = params.dig(:puzzle, :q) if params.dig(:puzzle, :q) != ''
+      if @q
+        @puzzles = Puzzle.search(@q, fields: [:name, :inventor, :producer],
+          misspellings: true).results.sort_by(&:name)
+        @puzzles += Puzzle.search(@q, fields: [:name, :inventor, :producer],
+          match: :word_middle, misspellings: false).results.sort_by(&:name)
+        @puzzles.uniq!
```

Comment:
Raczej się tak nie robi - robisz po dwa requesty do elastica i do bazy. Niewątpliwie da się jakoś lepiej skonfigurować te misspellings żeby Ci żadnych śmieci nie wrzucały - ale nie ma sensu żebyś się uczył aż tak dokładnie elastica teraz. Jeśli jesteś zadowolony z tego jak to działa to może tak zostać - ale jakbyś był u mnie w projekcie to na pewno bym Ci kazał to jakoś lepiej ogarnąć :-D